BigAnimal - Health Status doc (#5660)
EnterpriseDB/docs: product_docs/docs/biganimal/release/using_cluster/05_monitoring_and_logging/health_status.mdx

---
title: "Health Status"
---

> Health Status dashboard provides the status of the Postgres high-availability clusters. It provides status for both Primary/Standby and Distributed high-availability clusters.

Reviewer: ```suggestion
The Health Status dashboard provides real-time insight into the topology and health Postgres high-availability clusters. It supports both Primary/Standby and Distributed high-availability clusters.
```

> Health Status dashboard displays:

Reviewer: ```suggestion
The Health Status dashboard displays:
```

The **Health Status** tab displays the dashboard with health status categorized in sections:
- [Viewing Health Status dashboard](#viewing-health-status-dashboard)
  - [Global cluster health](#global-cluster-health)
  - [Regional nodes health and roles](#regional-nodes-health-and-roles)
  - [Replication status](#replication-status)

### Global cluster health

The global cluster health section displays the cluster-wide view including the following metrics:

- **Raft Status** (PGD only)&mdash; It indicates whether the Raft consensus algorithm is running correctly in the cluster. It verifies that one node is elected as the global leader and the Raft roles such as RAFT_LEADER and RAFT_FOLLOWER are defined correctly across all the nodes.
- **Replication Slot Status** (PGD only)&mdash; It indicates whether all the replication slots are in streaming state or not.
- **Clock Skew**&mdash; It indicates whether the node's clock is in sync and doesn't exceed a threshold of 60 seconds.
- **Proxy Status** (PGD only)&mdash; It provides the number of PGD proxies up and running as compared to the available proxies.
- **Node Status**&mdash; It provides the number of nodes that are up and running.

> - **Transaction rate**&mdash; It provides the total number of transactions including committed and rolled back transactions per second in the cluster.

Reviewer: On all of these lists, it would be more idiomatic (at least in American/British English) to omit the 'It' following the dash. Effectively the name of the label then functions as the subject of the sentence. We could even omit the dash and make this more explicit.

### Regional nodes health and roles

> The regional nodes health and roles section displays fine-grained health status at regional and node level. It is structured as an accordion, with each element representing a group of nodes deployed in the same region. Each item displays basic information like:

Reviewer: ```suggestion
The regional nodes health and roles section displays fine-grained health status at regional and node level. It is structured as an accordion, with each element representing a group of nodes deployed in the same region. Each item displays basic information including:
```

- **Proxy Status** (only PGD)&mdash; It indicates the number of active proxies compared to the available proxies in the specified regions.
- **Node Status** &mdash; The number of nodes up and running as compared to the available nodes in the specified region in a text chart. It provides the status of all nodes in the specified region using boolean indicator (green (OK)/red (KO)) in a text chart.

On expanding each item, it provides a list nodes with information like:
- Total number of active connections as compared to the maximum number of configured connections for each node.
- Memory usage percentage on a progress bar.

> - Storage usage percentage on a progress bar.

Reviewer 2: We recently added a "Node Ko" tag at node level that appears whenever the node is down. Can you please add this information?

Author: @aviola, can you share the screenshot of this?

For PGD, it provides the tags below the node name:
- **Raft Leader**&mdash; It indicates that the node is a Raft Leader locally in the region.
- **Raft Follower**&mdash; It indicates that the node is a Raft Follower locally in the region. 
- **Global Raft Leader**&mdash; It indicates that the nodes is a Raft Leader globally in the cluster.
- **Global Raft Follower**&mdash; It indicates that the node is a Raft Follower globally in the cluster.
- **Witness**&mdash; It indicates that the node is a witness in the PGD cluster. See [witness node docs](/pgd/latest/node_management/witness_nodes/) for more information.

For high-availability, it provides the tags below the node name:
- **Primary**&mdash; It indicates if the node role is primary.

### Replication status

The Replication Status section has a matrix displaying the replication lag across all the nodes of the cluster. The matrix provides different types of replication lags for **Write**, **Replay**, **Flush**, and **Sent**. It provides the lag in both bytes and time for **Write**, **Replay**, and **Flush** whereas only in bytes for **Sent**.

!!!note

> - In high-availability clusters, the replication occurs only from the primary(source) to replicas(target). So the matrix displays only one row for the source and multiple columns for the targets.

Reviewer: ```suggestion
- In high-availability clusters, replication occurs only from the primary (source) to the replicas (target). So the matrix displays only one row for the source and multiple columns for the targets.
```

> - In PGD clusters, the replications occurs bidirectional. So the matrix displays equal number of rows and columns (every node is both source and target).

Reviewer: ```suggestion
- In PGD clusters, the replications is bidirectional, so the matrix displays an equal number of rows and columns. Every node is both source and target.
```
